ir/marshal.go

```diff
@@ -0,0 +1,79 @@
+package ir
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+type MarshalType int
+
+// List of types
+const (
+	StringType = iota
```

Comment:
Use strings, not (implicitly generated) numbers. It's easier for forward compatibility.